docs/Project-Configuration/Containerized-Tasks.md

```diff
@@ -0,0 +1,118 @@
+
+Once containers are configured, they must be referenced by a build variant. Here is an example:
+
+``` yaml
+buildvariants:
+  - name: container-variant
+    display_name: Container Variant
+    run_on:
+      - example-container
+    tasks:
+      - name: test-graphql
+      - name: test-js
+```
+
+The container must be put in the `run_on` field, in the same way that
+a distro may be put there. Unlike the distro model, where a primary and
+secondary distro can be specified in this field (hence why the field is a list),
+only one container may be specified in the `run_on` field for a containerized
+variant.
+
+### UI Changes
+Once configured properly, a variant with container tasks is ready to schedule tasks.
+Once tasks get created, key differences to look for in Spruce are:
+
+#### Container Project Settings
+A new tab has been added to the project settings page for container configurations.
+Users can create a list of resource configuration presets that can be referenced via alias in the `size` field of their container YAML configurations.
+
+![containers.png](../images/containers.png)
+
+Options:
+
+-   `Name`: The alias for the resource preset. Names must be unique within the list.
+
+-   `Memory`: The amount of memory (in MiB) to allocate.
+
+-   `CPU`: The CPU units the container can use. These values are expressed in 'vCPU Units'. 1024 CPU units is the equivalent of 1vCPU.
+
+Users can define as many container configurations as needed, reflecting different appropriate resource needs for various tasks.
+
+#### Task Metadata
+A link to a container task's respective container replaces the typical host link. 
+
+![containerized_task_metadata.png](../images/containerized_task_metadata.png)
+
+#### Container Page
+
+The link in the task metadata sidebar takes you to the container page, which details the lifecycle of a container and their tasks. Like the host page, event logs exist charting the journey of a container task from initialization to termination.
+Task events such as container assignment and status changes are also recorded, as well as the clearing of a task from a container once it has run its course.
+
+![container_event_logs.png](../images/container_event_logs.png)
+
+### Disk Space Considerations
+Each instance is provisioned with 200GB of space; however, the actual disk space available for each container task can vary depending on the number and the nature of tasks sharing the same instance. The more memory and CPU a task requires, 
+the less likely it is to share an instance with other tasks. Conversely, less resource-hungry tasks are more likely to share the instance with other tasks, which means they will also be sharing the available disk space.
+
+In practical terms, this means that while each instance has a maximum of 200GB of disk space, please bear in mind that the effective disk space available to your container tasks might be less, depending on what other tasks are running concurrently on the same instance.
```

Comment:
If this is the case, I would say that a hard requirement of the users running container tasks is they are not allowed to run tasks that use excessive disk space (several GB or more). Sharing disk space between containers is a breach of the container isolation promise, which we should eventually find a better way to fix (if technically possible).